test/deployframework/local.go

```diff
@@ -0,0 +1,189 @@
+package deployframework
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	ansibleRunnerPath                   = "/tmp/ansible-operator/runner"
+	meteringOperatorName                = "metering-ansible-operator"
+	meteringOperatorLogName             = "metering-operator.log"
+	reportingOperatorLogName            = "reporting-operator.log"
+	destKubeConfigPath                  = "/kubeconfig"
+	reportingOperatorTLSDirName         = "reporting-operator-tls"
+	runReportingOperatorLocalScriptName = "run-reporting-operator-local.sh"
+	cleanupScriptName                   = "run-test-cleanup.sh"
+
+	enableDebug        = true
+	disableOCPFeatures = false
+)
+
+// LocalCtx holds all the necessary information to run e2e tests locally
+type LocalCtx struct {
+	Namespace                      string
+	BasePath                       string
+	KubeConfigPath                 string
+	HackScriptPath                 string
+	MeteringOperatorImage          string
+	ReportingAPIURL                string
+	MeteringOperatorCmd            *exec.Cmd
+	ReportingOperatorCmd           *exec.Cmd
+	RunReportingOperatorScriptPath string
+	Logger                         logrus.FieldLogger
+}
+
+// RunMeteringOperatorLocal is a method that runs the metering-operator locally
+func (lc *LocalCtx) RunMeteringOperatorLocal() error {
+	lc.MeteringOperatorCmd = exec.Command(
+		"docker", "run",
+		"--rm",
+		"-u", "0:0",
+		"--name", meteringOperatorContainerName,
+		"-v", ansibleRunnerPath,
+		"-v", fmt.Sprintf("%s:%s", lc.KubeConfigPath, destKubeConfigPath),
+		"-e", "KUBECONFIG="+destKubeConfigPath,
+		"-e", "OPERATOR_NAME="+meteringOperatorName,
+		"-e", "POD_NAME="+meteringOperatorName,
+		"-e", "WATCH_NAMESPACE="+lc.Namespace,
+		"-e", "ENABLE_DEBUG="+strconv.FormatBool(enableDebug),
+		"-e", "DISABLE_OCP_FEATURES="+strconv.FormatBool(disableOCPFeatures),
+		lc.MeteringOperatorImage,
+	)
+
+	lc.Logger.WithFields(logrus.Fields{
+		"command": lc.MeteringOperatorCmd.Args,
```

Comment:
I'm kind of thinking this shouldn't be a log field, but maybe just it's own debug log message.